### PR TITLE
Added preamble sections

### PR DESCRIPTION
**Pull request by jimblanc**
Adds the preamble sections to the RFC document including general purpose & library-wide enhancements.

### CODE REVIEW

File: RFC.md [hunk 1]
```diff
@@ -1,9 +1,18 @@
 # RFC: Amplify JS TypeScript Improvements
+Amplify JS is looking to improve our TypeScript support across the library to better serve our customers and provide a more intuitive & idiomatic developer experience. To this end we are requesting community feedback on a variety of changes and improvements that will be available to customers in the near future.
```
Comment (abdallahshaban557) [inline RFC.md]: To this end, we are requesting your feedback.

That will be available to you in our next major version release

Comment (elorzafe) [inline RFC.md]: ❤️

File: RFC.md [hunk 2]
```diff
@@ -1,9 +1,18 @@
...
 
-# General TypeScript Improvements
+This RFC is broken down into sections covering:
+- Library-wide TypeScript improvements
+- Changes to some of our core utilities such as `Hub` and Amplify configuration
+- Specific improvements to the `Auth`, `Storage`, and `API` categories and associated APIs
```
Comment (elorzafe) [inline RFC.md]: I would mention that we still want feedback for the other categories types as well

**Review by elorzafe — CHANGES_REQUESTED** [review]
Is looking great, I would still mention that we would like to get feedback on the categories we are not changing the APIs like `DataStore`

**Pull request merged**